### PR TITLE
Powerful23 dev

### PR DESCRIPTION
**Pull request by powerful23**
Unit tests for aws-amplify-react

For Federation identity: Add expiration time of the id_token

### CODE REVIEW

**Comment by powerful23** [comment]
Add Amazon Auth

**Comment by powerful23** [comment]
fix issues #86, #75, #71

**Comment by powerful23** [comment]
Federation login: Eliminate Warning when refreshing pages

**Comment by powerful23** [comment]
lint error fix

File: packages/aws-amplify-react/src/Auth/Provider/withFacebook.jsx [hunk 1]
```diff
@@ -39,18 +37,20 @@ export default function withFacebook(Comp) {
...
-            const { fb } = this.state;
+            const fb = window.FB;
             fb.api('/me', response => {
                 const user = {
                     name: response.name
                 }
 
-                Auth.federatedSignIn('facebook', accessToken, user)
+                Auth.federatedSignIn('facebook', { token: accessToken, expires_at }, user)
                     .then(crednetials => {
```
Comment (mlabieniec) [inline packages/aws-amplify-react/src/Auth/Provider/withFacebook.jsx]: typo here - should be credentials. For both google and facebook this line fails for me with

    cb=gapi.loaded_0:106 Uncaught TypeError: _awsAmplify.Auth.federatedSignIn is not a function
    at _class.federatedSignIn (withGoogle.js:83)
    at withGoogle.js:61
    at h.r2 (cb=gapi.loaded_0:114)
    at xs (cb=gapi.loaded_0:117)
    at Wq (cb=gapi.loaded_0:117)
    at _.C.uea (cb=gapi.loaded_0:116)
    at Ap (cb=gapi.loaded_0:110)
    at <anonymous>

File: packages/aws-amplify-react/src/Auth/Provider/withGoogle.jsx [hunk 2]
```diff
@@ -15,27 +15,25 @@ export default function withGoogle(Comp) {
...
             const profile = googleUser.getBasicProfile();
             const user = {
                 email: profile.getEmail(),
                 name: profile.getName()
             };
 
             const { onStateChange } = this.props;
-            return Auth.federatedSignIn('google', id_token, user)
+            return Auth.federatedSignIn('google', { token: id_token, expires_at }, user)
                 .then(crednetials => {
```
Comment (mlabieniec) [inline packages/aws-amplify-react/src/Auth/Provider/withGoogle.jsx]: similar to above type should be credentials and fails after federated login

    cb=gapi.loaded_0:106 Uncaught TypeError: _awsAmplify.Auth.federatedSignIn is not a function
    at _class.federatedSignIn (withGoogle.js:83)
    at withGoogle.js:61
    at h.r2 (cb=gapi.loaded_0:114)
    at xs (cb=gapi.loaded_0:117)
    at Wq (cb=gapi.loaded_0:117)
    at _.C.uea (cb=gapi.loaded_0:116)
    at Ap (cb=gapi.loaded_0:110)
    at <anonymous>

**Comment by powerful23** [comment]
Fix facebook login.
Add UI component analytics.
Add pre-commit configuration in package.json

**Comment by powerful23** [comment]
Coverage threshold set
Analytics event adjusted

**Pull request merged**